### PR TITLE
Fix date parsing

### LINKED ISSUE #2282: 'YYYY-MM' is interpreted as 'YYYY-MM-DD' --> narrow bars
Hey,

since upgrading to v0.16 I'm faced with the very skinny bar problem again, that happened a few versions back (for example #1558). 

I want to chart monthly values by grouping by month. I do that by using `to_char(timestamp, 'YYYY-MM')`.
Unfortunately for the resulting chart these values are interpreted as complete timestamp. Another problem is, that the visualized timestamps are cast to (I suppose) another timezone which means they are also off by one hour.

Example table:
image
Resulting chart:
image

If this is a bug report, please fill in the blanks:

*  I am using the Chrome browser.
*  My computer's OS is Mac.
*  I'm running Metabase on AWS
*  My database is postgres.
*  My Metabase version is 0.16.0.

### LINKED ISSUE #2288: 12th week repeated on x-axis
*  I am using the Chrome browser.
*  My computer's OS is El Capitan.
*  My database is PostgreSQL.
*  My Metabase version is v0.16.0.

When graphing a line or bar time series for This Year, grouped by week, if the browser window is larger than 1370px, the 12th week label is incorrectly duplicated on the x-axis:

screen shot 2016-04-01 at 11 44 34 am

screen shot 2016-04-01 at 11 44 15 am

screen shot 2016-04-01 at 9 19 45 am

### PR DESCRIPTION
**Pull request by tlrobinson**
Ok, I think this is a sane way to handle this. If we know a string has a `±hh:mm` timezone offset or `Z` suffix we use that in the parsing, otherwise we assume UTC.

Resolves #2282

Also removes using `utcOffset` in formatting, since timestamps from the same timezone may have different offsets if they span both DST and standard time.

Resolves #2288

### CODE REVIEW

File: frontend/src/lib/time.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+// only attempt to parse the timezone if we're sure we have one (either Z or ±hh:mm)
+// moment normally interprets the DD in YYYY-MM-DD as an offset :-/
+export function parseTimestamp(value) {
+    if (moment.isMoment(value)) {
+        return value;
+    } else if (typeof value === "string" && /(Z|[+-]\d\d:\d\d)$/.test(value)) {
+        return moment.parseZone(value);
+    } else {
+        return moment.utc(value);
```
Comment (agilliland) [inline frontend/src/lib/time.js]: what's the reason that this is necessary?  I would think that moment.parseZone() would always do what we want.

Comment (tlrobinson) [inline frontend/src/lib/time.js]: You would think so, but nope. https://github.com/moment/moment/issues/3083

**Comment by tlrobinson** [comment]
I think these tests are failing because I merged your branch in. Should be fixed when that's resolved and merged.

**Comment by agilliland** [comment]
k.  i'll do that once my branch is passing tests.

**Pull request merged**